src/components/ForumDiscussions/ForumPostCard.tsx

```diff
@@ -0,0 +1,325 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import React, { FC, useEffect, useState } from 'react';
+import { Topic } from './types';
+import { Divider } from 'antd';
+import { ClockCircleOutlined, LikeOutlined, EyeFilled } from '@ant-design/icons';
+import { ForumCommentsIcon } from '~src/ui-components/CustomIcons';
+import getRelativeCreatedAt from '~src/util/getRelativeCreatedAt';
+import Link from 'next/link';
+import getCategoryName from './utils/getCategoryName';
+import nextApiClientFetch from '~src/util/nextApiClientFetch';
+import ImageIcon from '~src/ui-components/ImageIcon';
+import formatAvatarUrl from './utils/FormatAvatarUrl';
+import SkeletonAvatar from '~src/basic-components/Skeleton/SkeletonAvatar';
+import SkeletonButton from '~src/basic-components/Skeleton/SkeletonButton';
+import { Pagination } from '~src/ui-components/Pagination';
+import { useTheme } from 'next-themes';
+import { LISTING_LIMIT } from '~src/global/listingLimit';
+
+interface ForumPostCardProps {
+	topics: Topic[];
+}
+
+function formatCount(count: number) {
+	if (count >= 10000) {
+		return (count / 1000).toFixed(1) + 'k';
+	}
+	return count.toString();
+}
+
+const ForumPostCard: FC<ForumPostCardProps> = ({ topics }) => {
+	// const [tagsModal, setTagsModal] = useState<boolean>(false);
+	const { resolvedTheme: theme } = useTheme();
+	const [usernames, setUsernames] = useState<{ [key: number]: string }>({});
+	const [userImg, setUserImg] = useState<{ [key: number]: string }>({});
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [currentPage, setCurrentPage] = useState<number>(1);
+	const itemsPerPage = LISTING_LIMIT;
+
+	const fetchUsernames = async (topicsToFetch: Topic[]) => {
+		setIsLoading(true);
+		const userMap: { [key: number]: string } = {};
+		const userImgMap: { [key: number]: string } = {};
+		for (const topic of topicsToFetch) {
```

Comment:
can't we get all topic once with these topic Id's

Reply:
i guess NO , we have to store data somewhere + Comments and Replies are also added by time so not a good idea by storing them